buildpacks/nodejs-npm-engine/src/errors.rs

```diff
@@ -0,0 +1,291 @@
+}
+
+fn on_inventory_parse_error(error: toml::de::Error, logger: Box<dyn StartedLogger>) {
+    print_error_details(logger, &error)
+        .announce()
+        .error(&formatdoc! {"
+            Failed to load available {npm} versions.
+
+            An unexpected error occurred while loading the available {npm} versions.
+        
+            {USE_DEBUG_INFORMATION_AND_RETRY_BUILD}
+
+            {SUBMIT_AN_ISSUE}
+        ", npm = fmt::value("npm") })
+}
+
+fn on_npm_version_resolve_error(requirement: Requirement, logger: Box<dyn StartedLogger>) {
+    logger.announce().error(&formatdoc! {"
+            Error resolving requested {npm} version {requested_version}.
+            
+            Can’t find the `npm` version that matches the requested version declared in {package_json} ({requested_version}).
+    
+            Verify that the requested version range matches a published version of {npm} by checking \
+            https://www.npmjs.com/package/npm?activeTab=versions or trying the following command:
+            
+                $ npm show 'npm@{requirement}' versions
+    
+            Update the {engines_key} field in {package_json} to a published {npm} version.
+    
+            {SUBMIT_AN_ISSUE}
+        ",
+        npm = fmt::value("npm"),
+        requested_version = fmt::value(requirement.to_string()),
+        package_json = fmt::value("package.json"),
+        engines_key = fmt::value("engines.npm")
+    })
+}
+
+fn on_npm_engine_layer_error(error: NpmEngineLayerError, logger: Box<dyn StartedLogger>) {
+    match error {
+        NpmEngineLayerError::Download(e) => {
+            print_error_details(logger, &e)
+                .announce()
+                .error(&formatdoc! {"
+                    Failed to download {npm}.
+
+                    An unexpected error occurred while downloading the {npm} package. This error can occur due to an unstable network connection.
+
+                    {USE_DEBUG_INFORMATION_AND_RETRY_BUILD}
+
+                    {SUBMIT_AN_ISSUE}
+                ", npm = fmt::value("npm") });
+        }
+        NpmEngineLayerError::OpenTarball(e) => {
+            print_error_details(logger, &e)
+                .announce()
+                .error(&formatdoc! {"
+                    Can’t open the downloaded {npm} package.
+
+                    An unexpected error occurred while opening the downloaded {npm} package file. 
+
+                    {USE_DEBUG_INFORMATION_AND_RETRY_BUILD}
+
+                    {SUBMIT_AN_ISSUE}
+                ", npm = fmt::value("npm") });
+        }
+        NpmEngineLayerError::DecompressTarball(e) => {
+            print_error_details(logger, &e)
+                .announce()
+                .error(&formatdoc! {"
+                    Can’t extract the downloaded {npm} package contents.
+
+                    An unexpected error occurred while extracting the contents of the downloaded {npm} package file.
```

Comment:
As above, I think this line is just a rewording of the line above it. Maybe delete one?